ros2/dynamic/src/SystemHandle.cpp

```diff
@@ -0,0 +1,632 @@
+                                        << std::endl;
+                                    is_success = false;
+                                    return;
+                                }
+
+                                // Check that the message name follows the ROS2 naming convention
+                                std::regex type_reg("[A-Z]([a-zA-Z0-9])+");
+                                if (!std::regex_match(aux, type_reg))
+                                {
+                                    logger_ << utils::Logger::Level::ERROR
+                                        << "The message name [" << aux
+                                        << "] doesn't follow the ROS2 naming convention."
+                                        << std::endl;
+                                    is_success = false;
+                                    return;
+                                }
+
+                                //TODO: Change when the services are implemented
+                                std::ofstream idlfile ("/tmp/" + mod.name() + "/msg/" + aux + ".idl");
+                                idlfile << idl << std::endl;
+                                idlfile.close();
+
+                                for (auto& mv_idl: custom_include_paths)
+                                {
+                                    std::string filename = std::filesystem::path(mv_idl).filename();
+                                    std::string path = std::filesystem::path(tmp / mod.name() / "msg" / filename);
+                                    logger_ << utils::Logger::Level::DEBUG
+                                        << "The file " << filename << " has been copied to " << path
+                                        << std::endl;
+                                    if (!std::filesystem::exists(path))
+                                    {
+                                        std::filesystem::copy(mv_idl, path);
+                                    }
+                                }
+
+                                std::string package_name = "--package_name " + mod.name();
+                                std::string path = "--install_path /opt/ros/" + ROS2_DISTRO;
+
+                                logger_ << is::utils::Logger::Level::INFO
+                                        << "Generating ROS2 Type Support for package: " << package_name
+                                        << std::endl;
+
+                                std::string command = "exec bash /tmp/generator.bash " + package_name + " " + path;
+
+                                if (ros2_modules.size() != 0)
+                                {
+                                    std::ostringstream stream;
+                                    std::copy(ros2_modules.begin(), ros2_modules.end(), std::ostream_iterator<std::string>(stream, ";"));
+                                    std::string depends = "--depedencies \"" + stream.str() + "\"";
```

Comment:
```suggestion
                                    std::string depends = "--dependencies \"" + stream.str() + "\"";
```

Does this mean that the bash script needs to be changed too?